app/grants/views.py

```diff
@@ -733,15 +731,22 @@ def get_grants_by_filters(
 
     if grant_regions:
         regions = grant_regions.split(',')
-        # 13. Fetch grants which have mentnioned regions
+        # 12. Fetch grants which have mentioned regions
         region_query= Q()
         for region in regions:
             region_query |= Q(region=region)
         _grants = _grants.filter(region_query)
 
+    if keyword:
+        # 13. Grant search by title & description
+        _grants = _grants.annotate(search=SearchVector('description'))
+        keyword_query = Q(title__icontains=keyword)
+        keyword_query |= Q(search=keyword)
+
+        _grants = _grants.filter(keyword_query)
 
```

Comment:
Is this enough to ensure that the sort order is respected? From our testing the paginated result was not being ordered by the corresponding `rank` - did you verify with enough test data?

Reply:
yes it works... i tested on staging... staging is currently on this branch... you could try it with "Save AWS" keyword

Reply:
> Is this enough to ensure that the sort order is respected?

sort is applied only after the search result is returned

> did you verify with enough test data?

yes on staging with "Save AWS" keyword (created multiple grants with title variations)...staging is currently on this branch